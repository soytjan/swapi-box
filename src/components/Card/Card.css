@import url("https://fonts.googleapis.com/css?family=Bungee+Hairline|Bungee+Outline|Roboto|Roboto+Condensed");
.card {
  text-transform: capitalize;
  background-color: #B5B5B5;
  box-shadow: 1px 1px 3px black;
  width: 100%;
  border-radius: 25px;
  min-height: 260px; }
  .card h4 {
    font-weight: 400; }
  .card li {
    list-style: none;
    text-align: left;
    line-height: 140%;
    display: flex;
    margin: 10px 0; }
  .card ul {
    padding: 0 15px; }
  .card .card-header {
    font-size: 1.1em;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 50px;
    padding: 10px 15px;
    border-radius: 8px 8px 0 0; }

.star-btn {
  min-height: 27px;
  background-image: url("./star-black.svg");
  min-width: 27px; }
  .star-btn:hover {
    background-image: url("./star-black-outline.svg");
    cursor: pointer; }

.attr {
  width: 50%;
  text-align: right; }

.value {
  font-family: "Roboto Condensed", sans-serif;
  width: 50%;
  padding-left: 5px; }

.people .card-header {
  background-color: #6A5671; }

.people .star-btn {
  background-color: #6A5671; }

.people .attr {
  color: #6A5671; }

.vehicles .card-header {
  background-color: #743B49; }

.vehicles .star-btn {
  background-color: #743B49; }

.vehicles .attr {
  color: #743B49; }

.planets .card-header {
  background-color: #41769F; }

.planets .star-btn {
  background-color: #41769F; }

.planets .attr {
  color: #41769F; }

.favorites .card-header {
  background-color: #5F8773; }

.favorites .star-btn {
  background-color: #5F8773; }

.favorites .attr {
  color: #5F8773; }

.favorited {
  background-color: #585858; }
  .favorited .attr {
    color: white; }
  .favorited .star-btn {
    background-image: url("./star-yellow.svg"); }
