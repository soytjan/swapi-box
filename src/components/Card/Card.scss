@import "../../variables.scss";

.card {
  text-transform: capitalize;
  background-color: $card-bg-color;
  box-shadow: 1px 1px 3px black;
  width: 100%;

  li {
    list-style: none;
    text-align: left;
    line-height: 150%;
  }

  ul {
    padding: 0 15px;
  }

  .card-header {
    font-size: 1.1em;
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 46px;
    padding: 10px 25px;
  }  
}

.star-btn {
  height: 27px;
  background-image: url('./star-black.svg');
  width: 27px; 

  &:hover {
    background-image: url('./star-black-outline.svg');
  }
}

.favorited {
  background-color: $card-active-bg-color;

  .star-btn {
    background-image: url('./star-yellow.svg');
  }
}

.attr {
  font-family: $sub-font;
}

.people {
  .card-header {
    background-color: $people-color;
  }

  .star-btn {
    background-color: $people-color;
  }
}

.vehicles {
  .card-header {
    background-color: $vehicles-color;
  }

  .star-btn {
    background-color: $vehicles-color;
  }
}

.planets {
  .card-header {
    background-color: $planets-color;
  }

  .star-btn {
    background-color: $planets-color;
  }
}