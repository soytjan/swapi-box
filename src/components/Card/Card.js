import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({info, type, onFavClick, isFavorited}) => {
  const attributes = Object.keys(info).filter((attr) => attr !== 'name');
  const klass = `card ${type} ${isFavorited}`;
  const listItems = attributes.map((attr, index) => {
    let value = info[attr];
    if (typeof value === 'object') {
      value = value.join(', ');
    }

    return (
      <li key={index}>
        <div className='attr'>{attr}:</div> 
        <div className='value'> {value}</div>
      </li>
    );
  });

  return ( 
    <article className={klass}>
      <div className='card-header'>
        <h4>{info.name}</h4>
        <button 
          onClick={() => onFavClick(info, isFavorited)} 
          className='star-btn'>
        </button>
      </div>
      <ul>
        {listItems}
      </ul>
    </article>
  );
};

Card.propTypes = {
  info: PropTypes.object,
  type: PropTypes.string,
  onFavClick: PropTypes.func,
  isFavorited: PropTypes.string
};

export default Card;