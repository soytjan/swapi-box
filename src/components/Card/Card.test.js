import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('expect to exist', () => {
    expect(Card).toBeDefined();
  })

  it.skip('should match snapshot', () => {
    const mockInfo = {name: "Coruscant", terrain: "cityscape, mountains", population: "1000000000000", climate: "temperate", residents: ["Finis Valorum", "Adi Gallia", "Jocasta Nu"]};
    const mockFavClick = jest.fn();
    const renderedComponent = shallow(<Card info={mockInfo} type='planets' isFavorited={false} onFavClick={mockFavClick} key={1} />)

    expect(renderedComponent).toMatchSnapshot();
  })
})