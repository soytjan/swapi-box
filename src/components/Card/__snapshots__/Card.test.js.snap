// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    info={
        Object {
            "climate": "temperate",
            "name": "Coruscant",
            "population": "1000000000000",
            "residents": Array [
              "Finis Valorum",
              "Adi Gallia",
              "Jocasta Nu",
            ],
            "terrain": "cityscape, mountains",
          }
    }
    isFavorited={false}
    onFavClick={[Function]}
    type="planets"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-header"
>
          <h4>
                    Coruscant
          </h4>
          <button
                    className="star-btn"
                    onClick={[Function]}
          />
</div>,
        <ul>
          <li>
                    terrain
                    : 
                    <span
                              className="attr"
                    >
                              cityscape, mountains
                    </span>
          </li>
          <li>
                    population
                    : 
                    <span
                              className="attr"
                    >
                              1000000000000
                    </span>
          </li>
          <li>
                    climate
                    : 
                    <span
                              className="attr"
                    >
                              temperate
                    </span>
          </li>
          <li>
                    residents
                    : 
                    <span
                              className="attr"
                    >
                              Finis Valorum, Adi Gallia, Jocasta Nu
                    </span>
          </li>
</ul>,
      ],
      "className": "card planets false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Coruscant
</h4>,
            <button
              className="star-btn"
              onClick={[Function]}
/>,
          ],
          "className": "card-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coruscant",
            },
            "ref": null,
            "rendered": "Coruscant",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "star-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              terrain
              : 
              <span
                            className="attr"
              >
                            cityscape, mountains
              </span>
</li>,
            <li>
              population
              : 
              <span
                            className="attr"
              >
                            1000000000000
              </span>
</li>,
            <li>
              climate
              : 
              <span
                            className="attr"
              >
                            temperate
              </span>
</li>,
            <li>
              residents
              : 
              <span
                            className="attr"
              >
                            Finis Valorum, Adi Gallia, Jocasta Nu
              </span>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "terrain",
                ": ",
                <span
                  className="attr"
>
                  cityscape, mountains
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "terrain",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "cityscape, mountains",
                  "className": "attr",
                },
                "ref": null,
                "rendered": "cityscape, mountains",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "population",
                ": ",
                <span
                  className="attr"
>
                  1000000000000
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "population",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1000000000000",
                  "className": "attr",
                },
                "ref": null,
                "rendered": "1000000000000",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "climate",
                ": ",
                <span
                  className="attr"
>
                  temperate
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "climate",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "temperate",
                  "className": "attr",
                },
                "ref": null,
                "rendered": "temperate",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "residents",
                ": ",
                <span
                  className="attr"
>
                  Finis Valorum, Adi Gallia, Jocasta Nu
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "residents",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Finis Valorum, Adi Gallia, Jocasta Nu",
                  "className": "attr",
                },
                "ref": null,
                "rendered": "Finis Valorum, Adi Gallia, Jocasta Nu",
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header"
>
            <h4>
                        Coruscant
            </h4>
            <button
                        className="star-btn"
                        onClick={[Function]}
            />
</div>,
          <ul>
            <li>
                        terrain
                        : 
                        <span
                                    className="attr"
                        >
                                    cityscape, mountains
                        </span>
            </li>
            <li>
                        population
                        : 
                        <span
                                    className="attr"
                        >
                                    1000000000000
                        </span>
            </li>
            <li>
                        climate
                        : 
                        <span
                                    className="attr"
                        >
                                    temperate
                        </span>
            </li>
            <li>
                        residents
                        : 
                        <span
                                    className="attr"
                        >
                                    Finis Valorum, Adi Gallia, Jocasta Nu
                        </span>
            </li>
</ul>,
        ],
        "className": "card planets false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Coruscant
</h4>,
              <button
                className="star-btn"
                onClick={[Function]}
/>,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coruscant",
              },
              "ref": null,
              "rendered": "Coruscant",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "star-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                terrain
                : 
                <span
                                className="attr"
                >
                                cityscape, mountains
                </span>
</li>,
              <li>
                population
                : 
                <span
                                className="attr"
                >
                                1000000000000
                </span>
</li>,
              <li>
                climate
                : 
                <span
                                className="attr"
                >
                                temperate
                </span>
</li>,
              <li>
                residents
                : 
                <span
                                className="attr"
                >
                                Finis Valorum, Adi Gallia, Jocasta Nu
                </span>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "terrain",
                  ": ",
                  <span
                    className="attr"
>
                    cityscape, mountains
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "terrain",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cityscape, mountains",
                    "className": "attr",
                  },
                  "ref": null,
                  "rendered": "cityscape, mountains",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "population",
                  ": ",
                  <span
                    className="attr"
>
                    1000000000000
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "population",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1000000000000",
                    "className": "attr",
                  },
                  "ref": null,
                  "rendered": "1000000000000",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "climate",
                  ": ",
                  <span
                    className="attr"
>
                    temperate
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "climate",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "temperate",
                    "className": "attr",
                  },
                  "ref": null,
                  "rendered": "temperate",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "residents",
                  ": ",
                  <span
                    className="attr"
>
                    Finis Valorum, Adi Gallia, Jocasta Nu
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "residents",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Finis Valorum, Adi Gallia, Jocasta Nu",
                    "className": "attr",
                  },
                  "ref": null,
                  "rendered": "Finis Valorum, Adi Gallia, Jocasta Nu",
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
